### PR TITLE
Convert to graphql

### PR DESCRIPTION
**Pull request by chrpeter**

### CODE REVIEW

File: src/containers/SubjectPage/SubjectPage.jsx [hunk 1]
```diff
@@ -20,69 +22,82 @@ import {
...
     const { subjectId } = getUrnIdsFromProps(ctx);
-    fetchSubjects();
-    fetchSubjectFilters(subjectId);
-    fetchTopicsWithIntroductions({ subjectId });
+    const urlParams = queryString.parse(location.search || '');
+    try {
+      return runQueries(client, [
+        {
+          query: subjectQuery,
+          variables: { subjectId, filterIdss: urlParams.filters || '' },
```
Comment (oyvinmar) [inline src/containers/SubjectPage/SubjectPage.jsx]: `filterIds` er vel det du mener?

**Comment by chrpeter** [comment]
Lagt til dette nå @oyvinmar

File: src/queries.js [hunk 2]
```diff
@@ -127,13 +130,23 @@ export const articleInfoFragment = gql`
...
 export const subjectQuery = gql`
   ${topicInfoFragment}
-  query subjectQuery($subjectId: String!) {
+  query subjectQuery($subjectId: String!, $filterIds: String) {
     subject(id: $subjectId) {
       id
       name
       path
-      topics(all: true) {
+      topics(all: true, filterIds: $filterIds) {
```
Comment (oyvinmar) [inline src/queries.js]: mm, we should not use both all and subtopics here. Should probably only use all and change make function similar to getSubjectMenu selector.

File: src/shapes.js [hunk 3]
```diff
@@ -91,17 +91,15 @@ export const FootNoteShape = PropTypes.shape({
   publisher: PropTypes.string.isRequired,
 });
 
-export const GraphqlErrorShape = PropTypes.shape({
```
Comment (oyvinmar) [inline src/shapes.js]: Why is this removed?

File: src/containers/SubjectPage/SubjectPage.jsx [hunk 4]
```diff
@@ -121,7 +140,7 @@ class SubjectPage extends Component {
                   </h1>
                   <TopicIntroductionList
                     toTopic={toTopic(subjectId)}
-                    topics={subjectTopics}
+                    topics={topics}
```
Comment (oyvinmar) [inline src/containers/SubjectPage/SubjectPage.jsx]: Since subjectQuery uses `all` param. Every topic (including subtopics) is in the topic list.

File: src/shapes.js (hunk 3 above)
Comment (chrpeter) [reply src/shapes.js]: Moved to GraphqlShapes.

**Comment by chrpeter** [comment]
Should now be updated to handle filter equal in both the menu and the subjectpage.
